### PR TITLE
:rainbow: details_model_record field refactor

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

**Review by ilmir-k — APPROVED** [review]

**Comment by ilmir-k** [comment]
@KolushovAlexandr change the version please

File: base_details/models/base_details.py [hunk 1]
```diff
@@ -21,19 +29,36 @@ def details(self):
             details_record = None
         return details_record
 
-    details_model_record = fields.Reference(selection="_model_selection", string='Record')
-    details_model = fields.Char(compute="_compute_details", string='Detail Model', store=True)
-    details_res_id = fields.Integer(compute="_compute_details", string='Details', store=True)
-    details_model_exists = fields.Boolean(compute="_compute_details", string='Details Model Exists', store=True)
+    details_model = fields.Selection(selection="_model_selection", string='Model', store=True)
+    details_model_record = fields.Reference(selection="_compute_details_model_record", string='Record')
```
Comment (yelizariev) [inline base_details/models/base_details.py]: Don't use ``_compute`` prefix for selection

File: base_details/models/base_details.py [hunk 2]
```diff
@@ -21,19 +29,36 @@ def details(self):
...
+            if rec.details_model and rec.details_model in [x.model for x in rec.env['ir.model'].search([])]:
+                rec.details_model_exists = True
+            else:
+                rec.details_model_exists = False
 
     @api.multi
-    @api.depends('details_model_record')
-    @api.onchange('details_model_record')
-    def _compute_details(self):
+    @api.onchange('details_model_exists', 'details_model_record')
```
Comment (yelizariev) [inline base_details/models/base_details.py:53]: Why it's ``onchange`` and not ``api.depends``?

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: base_details/models/base_details.py (hunk 2 above)
Comment (KolushovAlexandr) [reply base_details/models/base_details.py:53]: @yelizariev Because it should be updated right on the form

**Review by yelizariev — APPROVED** [review]

**Pull request merged**